cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/AbstractMapReduceContextBuilder.java

```diff
@@ -63,6 +63,7 @@ public BasicMapReduceContext build(MapReduceMetrics.TaskType type,
                                      String taskId,
                                      long logicalStartTime,
                                      String workflowBatch,
+                                     String adapterName,
```

Comment:
It is probably better to add new parameter in the constructor at the end rather than stick it in the middle. Make it easier to avoid merge conflict and update code.